### PR TITLE
[c++] Create helper function for casting bits to uint8

### PR DESCRIPTION
**Pull request by nguyenv**
Avoid duplicating code by creating helper function `SOMAArray::_cast_bit_to_uint8`

### CODE REVIEW

File: libtiledbsoma/src/soma/soma_array.h [hunk 1]
```diff
@@ -950,6 +950,9 @@ class SOMAArray : public SOMAObject {
         }
     }
 
+    // Help function to cast Boolean of bits (Arrow) to uint8 (TileDB)
```
Comment (johnkerl) [inline libtiledbsoma/src/soma/soma_array.h:953]:
```suggestion
    // Helper function to cast Boolean of bits (Arrow) to uint8 (TileDB)
```

**Review by johnkerl — CHANGES_REQUESTED** [review]
Did you perhaps forget to `git add` the file implementing `_cast_bit_to_uint8`? I don't see it ...

File: libtiledbsoma/src/soma/soma_array.cc [hunk 2]
```diff
@@ -607,6 +607,42 @@ ArrowTable SOMAArray::_cast_table(
...
+        data = arrow_array->buffers[1];
+    }
+
+    auto sz = arrow_array->length;
+
+    std::vector<uint8_t> casted;
+    for (int64_t i = 0; i * 8 < sz; ++i) {
+        uint8_t byte = ((uint8_t*)data)[i];
+        for (int64_t j = 0; j < 8; ++j) {
+            casted.push_back((uint8_t)((byte >> j) & 0x01));
```
Comment (johnkerl) [inline libtiledbsoma/src/soma/soma_array.cc:625]: It's easy (in my years of doing things like this) to get the bit-ordering backward.

Let's make sure we have focused unit-test coverage in `libtiledbsoma/test/unit_*.cc`

**Comment by nguyenv** [comment]
Included in #2786.

**Pull request closed without merging**